app/_kong_plugins/tls-metadata-headers/index.md

```diff
@@ -38,4 +38,41 @@ search_aliases:
   - certificates
 ---
 
-## Overview
+The TLS Metadata Header plugin detects client certificates in requests, extracts the TLS metadata (such as the URL-encoded client certificate), and injects this metadata into HTTP headers. It does not validate client certificates.
+
+Here are some use cases where the TLS Metadata Header plugin can be helpful:
+* Pass TLS client certificate metadata to an upstream service, enabling it to perform validation of the proxied certificate
+* Establish trust between microservices and use the extracted metadata to route requests differently based on the client’s certificate metadata (for example, different routes for different departments or services)
+* Enforce access control based on certain attributes of the client certificate, like the client’s organization (extracted from the certificate's subject DN)
+* Log or audit extracted metadata from client certificates
+
+{:.warning}
+> **Important:** This plugin **must** be used in conjunction with another plugin that requests a client certificate, such as the [mTLS Authentication](/plugins/mtls-auth/) or [TLS Handshake Modifier](/plugins/tls-handshake-modifier/) plugins. 
+
+## How it works
+
+The following explains how the TLS Metadata Headers plugin works:
+
+1. A client request is sent with a TLS certificate on a Gateway Service.
+2. Either the mTLS Authentication or TLS Handshake Modifier plugin is configured on the Gateway Service, making the client certificate available to other plugins.
+3. The TLS Metadata Header plugin, configured on the same Gateway Service, accesses the client certificate and extracts metadata such as the certificate itself, serial number, issuer Distinguished Name (DN), subject DN, and SHA1 fingerprint. It also retrieves the full client certificate chain.
+4. If `inject_client_cert_details` is enabled, the TLS Metadata Header plugin injects the extracted TLS client certificate metadata into HTTP headers.
+
+{% mermaid %}
+sequenceDiagram
+    actor C as Client
+    participant M as mTLS Authentication/TLS Handshake Modifier plugin
+    participant T as TLS Metadata Header plugin
+    participant H as Upstream service
+
+    C->>M: Sends request with TLS certificate
+    M->>M: Makes client certificate available to other plugins
+    M->>T: Extracts metadata: <br>certificate, serial number, issuer DN, <br>subject DN, SHA1 fingerprint, certificate chain
+    T->>T: Checks if `inject_client_cert_details` is enabled
+    alt `inject_client_cert_details` is enabled
+        T->>H: Injects metadata into HTTP headers
+    else `inject_client_cert_details` is disabled
+        T->>T: Does nothing
+    end
```

Comment:
```suggestion
    C->>M: Sends request with <br>TLS certificate
    M->>M: Makes client certificate <br>available to other plugins
    M->>T: Extracts metadata
    T->>T: Checks if inject_client_cert_details <br>is enabled
    alt inject_client_cert_details is enabled
        T->>H: Injects metadata into HTTP headers
    else inject_client_cert_details is disabled
        T->>T: Does nothing
    end
```
Detailing the specific metadata is too much info for the diagram. Also, mermaid doesn't know what to do with backticks.

Also adding some linebreaks to make this diagram less wide and easier to read.